tests/system/test_multigroups/test_multigroups.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (C) 2015-2022, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import hashlib
+import os
+import random
+import string
+from time import sleep
+
+import pytest
+
+from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.tools.system import HostManager
+
+# Test variables
+
+test_hosts = ['wazuh-master', 'wazuh-worker1', 'wazuh-worker2']
+agent_groups = {'wazuh-agent1': ['default', 'test_mg_0'],
+                'wazuh-agent2': ['default', 'test_mg_1']}
+
+shared_folder_path = os.path.join(WAZUH_PATH, 'etc', 'shared')
+mg_folder_path = os.path.join(WAZUH_PATH, 'var', 'multigroups')
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'provisioning',
+                              'basic_cluster', 'inventory.yml')
+hm = HostManager(inventory_path)
+token = hm.get_api_token(test_hosts[0])
+time_to_update = 10
+time_to_sync = 20
+
+
+# Functions
+
+def calculate_mg_name(groups_list):
+    """Determine multigroup name from a list of groups.
+
+    Args:
+        groups_list (list): List of groups that form a multigroup.
+
+    Returns:
+        str: Multigroup name.
+    """
+    return hashlib.sha256(','.join(groups_list).encode()).hexdigest()[:8]
+
+
+def get_mtime(folders_to_check, hosts=None):
+    """Get modification time of each file within the listed directories, for each listed host.
+
+    Args:
+        folders_to_check (iterable of str): Directories to recursively traverse to get the mtime of their files.
+        hosts (list): List of hosts whose folders should be iterated.
+
+    Returns:
+        result (dict): Dictionary with the host, path and modification time of each file.
+    """
+    hosts = test_hosts if hosts is None else hosts
+    result = {}
+
+    for host in hosts:
+        result[host] = {}
+        for folder in folders_to_check:
+            result[host].update(
+                {file['path']: file['mtime'] for file in hm.find_file(host, folder, recurse=True)['files']}
+            )
+
+    return result
+
+
+def get_agent_id(token, agent_name):
+    """Get agent ID from its name.
+
+    Args:
+        token (str): API token.
+        agent_name (str): Agent name.
+
+    Returns:
+        str: Agent ID.
+    """
+    agent_id = hm.make_api_call(host=test_hosts[0], token=token, method='GET', endpoint=f'/agents?name={agent_name}')
+    assert agent_id['status'] == 200, f'Failed trying to get ID of agent {agent_name}'
```

Comment:
Check the quotes in this file, as sometimes they do not follow the [style guide](https://github.com/wazuh/wazuh-qa/wiki/Development-guidelines#style) of this repo. 
```suggestion
    assert agent_id['status'] == 200, f"Failed trying to get ID of agent {agent_name}"
```